### DISCUSSION TITLE
Non-linear model

### DISCUSSION THREAD

**Discussion by mia-K71** [IRIS-Solutions-Team/IRIS-Toolbox#355] [Q&A]
Hi,

I am trying to build a simple balance sheet model in IRIS. In the model  foreign assets (FA) are represented as follows:

FA = FA{-1} + INC + INT; where INC - income from exchange rate changes and interest income, INT - interventions. 
 
INC = (yield + dexchange - 2 )*FA{-1}

INT = res_int;

I model yield and exchange rate changes as AR process (log(yield) = a*log(yield{-1})+ (1-a)*log(ss_yield)+res_yield !! yield = ss_yield and log(dexchange) = a*log(dexchange{-1})+ (1-a)*log(ss_dexchange)+ res_dexchange !! dexchange = ss_dexchange).

I declare INT (difference between bought and sold currency ) as a log_variable as it can be negative.

However, I am fail to solve this model because of steady state issues.  Could you point out what mistakes I'm making? Thank you.

Best, 
Mia

**Comment by jaromir-benes** [comment]
HI Mia

The rules for log-linear variables are the following

#1 If a variable is growing at a constant rate in steady state (rate of
change), it *must* be declared as log-variables

#2 if a variable is growing at a constant first difference in steady state,
it *must not* be declared as a log variables

#3 if a variable is flat in steady state (either stationary or a unit root
variables with no  growth) you can choose

#4 if a variable can flip its sign, it *must not* be declared as a log
variable

#5 If there is a conflict between #1 and #4, you cannot have that variable
directly in the model, and you must create a suitable transformation of
that variable, e.g. a ratio over GDP.

I guess your case may fall under #5 :)

Let me know if this helps.

Best
Jaromir

**Reply by mia-K71** [reply]
Dear Jaromir, 

Thank you for such a quick answer. I want to clarify what I wrote above. I declare INT in the block log_variable !!all but INT.

 I tried to model INT as a ratio to GDP and then in the equation of FA put INT_NGDP*NGDP. However, this doesn't help. Maybe I missed something.

Best,
Mia
